File: go_files/Ex5.go
package main

import (
  "fmt"
)


type Animal interface {
  Som() string
}

type Cachorro struct{}

func (c Cachorro) Som() string {
  return "Au au!"
}

type Gato struct{}

func (g Gato) Som() string {
  return "Miau!"
}

type Vaca struct{}

func (v Vaca) Som() string {
  return "Muu!"
}

func FazerSom(animais []Animal) {
  for _, animal := range animais {
    fmt.Println(animal.Som())
  }
}

func main() {

  cachorro := Cachorro{}
  gato := Gato{}
  vaca := Vaca{}
  animais := []Animal{cachorro, gato, vaca}

  FazerSom(animais)
}

File: pyton_files/ex5.py
class Animal:
    def __init__(self, nome):
        self.nome = nome
    
    def som(self):
         raise NotImplementedError("Metodo som não implementado")

class Cachorro(Animal):
    def som(self):
        return f"{self.nome} Au au!"

class Gato(Animal):
    def som(self):
        return f"{self.nome} Miau!"


def emitir_sons(animais):
    for animal in animais:
        print(animal.som())


meu_cachorro = Cachorro("Dogpoll")
meu_gato = Gato("Jiji")
outro_gato = Gato("Lily")

animais = [meu_cachorro, meu_gato, outro_gato]
emitir_sons(animais)